src/compile/model.ts

```diff
@@ -229,7 +239,8 @@ export abstract class Model {
     this.renameTopLevelLayoutSize();
 
     this.parseSelection();
-    this.parseData(); // (pathorder) depends on markDef; selection filters depend on parsed selections.
+    this.parseProjection();
+    this.parseData(); // (pathorder) depends on markDef; selection filters depend on parsed selections; depends on projection.
```

Comment:
> depends on projection.

which part of data depends on projection? 

(Other text on this line is a bit more specific)